adr/0019-GPIO.md

```diff
@@ -0,0 +1,31 @@
+# 0019. GPIO
+
+Date: 2021-12-14
+
+## Status
+
+Accepted
+
+## Context
+
+We currently have a lot of integrations that use or are based on GPIO (including SPI and i2c busses).
+
+- All of these integrations have a low usage count (source Home Assistant Analytics),
+  but it adds the same amount of maintenance and review time to the core.
+- Most of these integrations are unmaintained.
+- The use of GPIO via containers (including our Home Assistant OS, Container
+  & Supervised installation methods) is often unstable or complicated.
+- Support for GPIO on specific platforms/SBCs: mileage varies (e.g., Raspberry Pi on 64 bits platforms).
+- The use of these sensors, in general, is better supported and done via dedicated projects like ESPHome.
+
+## Decision
+
+- We no longer accept integrations that integrate with devices over GPIO.
+- We identify, deprecate and remove integrations that use GPIO.
+- It will still be possible to have custom integrations that use GPIO.
+
```

Comment:
Does this need a comment on whether HAOS will continue to provide and support a means of accessing GPIO in the long term

(Not sure if this applies but I assume there is some effort in HAOS to provide reliable pass through of those pins)

Reply:
I dunno, I've kept that out of scope for this ADR (which is focused solely on the integrations parts). I don't see a reason to remove support from that 🤷